peps/pep-0807.rst

```diff
@@ -0,0 +1,398 @@
+containing a JSON object with the following field:
+
+- ``audience``: a string containing the expected OIDC audience.
+
+On failure, the server **MUST** respond with any standard HTTP
+error code in the 400 or 500 range to indicate an error condition.
+Failure responses **MAY** include a body which, if present,
+**MUST** be a JSON object containing an
+:ref:`Error Response <error-responses>`.
+
+.. _token-minting:
+
+Token Minting
+~~~~~~~~~~~~~
+
+After the uploading client has performed
+:ref:`audience retrieval <audience-retrieval>` and obtained an
+identity credential from the Trusted Publishing provider, it can
+proceed to mint an upload credential.
+
+To mint an upload credential, the uploading client performs
+an HTTP POST request to the *mint token endpoint* obtained during
+:ref:`discovery <discovery>`.
+
+On success, the server responds with a ``200 OK`` status code and a body
+containing a JSON object with the following field:
+
+- ``token``: a string containing the upload credential. The format
+  of the upload credential is implementation-defined and index-specific.
+- ``expires``: an **optional** integer containing a Unix timestamp
+  indicating when the upload credential expires. If this field is not
+  present, the uploading client **MAY** assume an expiration point
+  of not more than 15 minutes (900 seconds) after the time of
+  their request.
+
+  The server **MUST NOT** issue temporary upload credentials
+  that expire in less than 15 minutes (900 seconds) or more than
+  6 hours (21,600 seconds) from the time of the request. [#fn-expires]_
+
+  The uploading client **MAY** use this time (or the minimum specified
+  above) to determine when to refresh the upload credential, if needed.
+
+On failure, the server **MUST** respond with any standard HTTP
+error code in the 400 or 500 range to indicate an error condition.
+Failure responses **MUST** include a body which, if present,
+**MUST** be a JSON object containing an :ref:`Error Response <error-responses>`.
+
+.. _error-responses:
+
+Error Responses
+---------------
+
+When an error response body is included, it **MUST** be a JSON object
+containing the following fields:
+
+- ``message``: a string containing a short, high-level
+    human-readable summary of the error.
+
+- ``errors``: an array of one or more objects, each containing
+    the following fields:
+
+    - ``code``: a string containing a machine-readable error code.
+    - ``description``: a string containing a human-readable
+      description of the error.
+
+This PEP does not specify any particular error codes. Clients **SHOULD NOT**
+assume that error codes are consistent across different indices, and instead
+**MUST** treat error codes as opaque strings.
+
+Security Implications
+=====================
+
+This PEP does not identify any positive or negative security implications
+associated with the Trusted Publishing discovery or exchange flows themselves.
```

Comment:
From the Motivation, this PEP seeks to improve security / transparency by formally specifying the Trusted Publishing process. Worth mentioning that here?

Reply:
Yeah, good call, will add.